### ISSUE TITLE
[ENH]: use_holdout argument for get_leaderboard() function in regression and classification

### ISSUE THREAD

**Issue by Ali-Flt** [pycaret/pycaret#3204]
### Describe the feature you want to add to this project

If I've understood correctly right now the `get_leaderboard()` function returns metrics (MSE, RMSE, R2, ...) based on the train dataset. It would be useful to have the option to return metric values based on the holdout set.

### Describe your proposed solution

`use_holdout : bool` argument for `get_leaderboard()` function

**Comment by Ali-Flt** [comment]
as a workaround I've used this code to get the score I want for the holdout set. Am I doing it right?
```
from sklearn.metrics import mean_absolute_error
leader_board = get_leaderboard()
  for i, model in leader_board.iterrows():
      predict_result = predict_model(model['Model'])
      mae = mean_absolute_error(y_true=predict_result['label'], y_pred=predict_result['prediction_label'], verbose=False) 
      leader_board.loc[i, 'test_MAE'] = mae

  leader_sorted = leader_board.sort_values(by='test_MAE', ascending=True)
  best_model = leader_sorted.iloc[0]['Model']
```

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
